File: rnamake/unittests/segmenter_unittest.py
import unittest
import rnamake.segmenter
import rnamake.pose
import rnamake.settings
import rnamake.pose_factory as pf
import rnamake.motif_type as motif_type

class SegmenterUnittest(unittest.TestCase):

    def test_creation(self):
        s = rnamake.segmenter.Segmenter()

    def _test_apply(self):
        path = rnamake.settings.UNITTEST_PATH + "/resources/motifs/p4p6"
        p = pf.factory.pose_from_file(path)
        twoways = p.motifs(motif_type.TWOWAY)
        for i, t in enumerate(twoways):
            s = rnamake.segmenter.Segmenter()
            segments = s.apply(p, t.ends)
            self.failUnless(len(t.residues()) == len(segments.removed.residues()))

        nways = p.motifs(motif_type.NWAY)
        for i, t in enumerate(nways):
            s = rnamake.segmenter.Segmenter()
            segments = s.apply(p, t.ends)
            self.failUnless(len(t.residues()) == len(segments.removed.residues()))

    def test_apply_2(self):
        path = rnamake.settings.UNITTEST_PATH + "/resources/motifs/p4p6"
        p = pf.factory.pose_from_file(path)
        ends = [p.get_basepair(name="A112-A208")[0],
                p.get_basepair(name="A128-A194")[0]]
        #print ends
        s = rnamake.segmenter.Segmenter()
        segments = s.apply(p, ends)
        m = segments.remaining





def main():
    unittest.main()

if __name__ == '__main__':
    main()

File: rnamake/lib/RNAMake/unittests/motif_unittests/segmenter_unittest.cpp
//
//  segmenter_unittest.cpp
//  RNAMake
//
//  Created by Joseph Yesselman on 9/17/15.
//  Copyright (c) 2015 Joseph Yesselman. All rights reserved.
//

#include "segmenter_unittest.h"
#include "motif/pose_factory.h"
#include "motif_tools/segmenter.h"

int
SegmenterUnittest::test_creation() {
    /*Segmenter s;
    PoseFactory pf;
    String path = base_dir() + "/rnamake/unittests/resources/motifs/p4p6/p4p6.pdb";
    auto p = pf.pose_from_file(path);
    auto bp1 = p->get_basepair("A112-A208")[0];
    auto bp2 = p->get_basepair("A132-A191")[0];
    s.apply(p, BasepairOPs { bp1, bp2} );*/
    return 1;
}

int
SegmenterUnittest::run() {
    if (test_creation() == 0)        { std::cout << "test_creation failed" << std::endl;  }
    return 1;
}